runtime/src/rpc.rs

```diff
@@ -70,12 +72,30 @@ impl PolkaBtcProvider {
 
     /// Gets a copy of the signer with a unique nonce
     async fn get_unique_signer(&self) -> PolkaBtcSigner {
-        // TODO: refresh from account store
         let mut signer = self.signer.write().await;
-        // return the current value, increment afterwards
-        let ret = signer.clone();
-        signer.increment_nonce();
-        ret
+        let mut account_nonce = self.nonce.load(Ordering::SeqCst);
+
+        // For getting the nonce, use latest, possibly non-finalized block.
+        // TODO: we might want to wait until the latest block is actually finalized
+        // query account info in order to get the nonce value used for communication
+        let system_nonce = crate::frame_system::AccountStoreExt::account(
+            &self.ext_client,
+            self.account_id.clone(),
+            Option::<H256>::None,
+        )
+        .await
+        .map_or_else(|_| u32::MIN, |info| info.nonce);
+
+        account_nonce = std::cmp::max(system_nonce, account_nonce);
+
+        let mut cloned_signer = signer.clone();
+        cloned_signer.set_nonce(account_nonce);
+
+        account_nonce = account_nonce.wrapping_add(1);
```

Comment:
Did you check if wrapping will work? If this is supported, I am surprised that `increment_nonce` is not doing this already